tools/amp_segment/ina_speech_carbonate.py

```diff
@@ -0,0 +1,192 @@
+            rc = int(stdout[-1])
+            logger.info(f"Return code: {rc}")
+            logger.debug(f"STDOUT: {''.join(stdout)}")
+            logger.debug(f"STDERR: {''.join(stderr.readlines())}")
+
+            if rc != 0:
+                raise Exception(f"Cannot queue job RC={rc}")
+
+            # the job is queued, so we just need to wait for the finished.out
+            # file to appear (which contains the return code)
+            poll = 30
+            logger.info(f"Checking every {poll}s for {workspace}/finished.out to appear")
+            while True:
+                try:
+                    sftp.stat(f"{workspace}/finished.out")
+                    logging.info(f"{workspace}/finished.out has appeared")
+                    break
+                except FileNotFoundError as e:
+                    # we're expecting this.
+                    logging.debug(f"Still waiting.")
+                    time.sleep(poll)
+
+            with tempfile.TemporaryDirectory() as td:
+                logging.debug("Retrieving finished.out")
+                sftp.get(f"{workspace}/finished.out", f"{td}/finished.out")
+                with open(f"{td}/finished.out") as f:
+                    rc = int(f.readline())
+                if rc != 0:
+                    # uh oh.  Grab the stdout and stderr bits and dump them.
+                    logging.error(f"Not-zero return code from batch job: {rc}")
+                    for out in ("stdout.txt", "stderr.txt"):
+                        try:
+                            sftp.get(f"{workspace}/{out}", f"{td}/{out}")
+                            with open(f"{td}/{out}") as f:
+                                output = f.read()
+                            logging.error(f"Contents of {out}:\n{output}")
+                        except Exception as e:
+                            pass
+                    error = True
+                else:
+                    for o in out_map:
+                        logging.info(f"Retrieving {o} as {out_map[o]}")
+                        sftp.get(f"{workspace}/{o}", out_map[o])
+                        
+        except Exception as e:
+            logger.critical(f"Error during processing: {e}")
+            error = True
+        finally:
+            if args.nocleanup:
+                logger.info(f"Not cleaning up remote directory {workspace}")
+            else:
+                logger.info(f"Cleaning up work directory {workspace}")
+                sftp.chdir("..")
+                clean_up(sftp, workdir)
+
+    exit(1 if error else rc)
+
+
+def clean_up(sftp, dir):
+    entries = list(sftp.listdir_iter(dir))
+    for f in entries:
+        if f.st_mode & stat.S_IFDIR:
+            clean_up(sftp, f"{dir}/{f.filename}")
+        else:
+            logger.debug(f"Removing {dir}/{f.filename}")
+            sftp.remove(f"{dir}/{f.filename}")
+    logger.debug(f"Removing directory {dir}")
+    sftp.rmdir(dir)
+
+if __name__ == "__main__":
+    main()
```

Comment:
There seems to be lots of repetition between the 2 HPC related scripts. Can we extract the logic into some common py file that can be shared by all hpc tools?